nexusclient/client.go

```diff
@@ -0,0 +1,725 @@
+package nexusclient
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log/slog"
+	"net/http"
+	"net/url"
+	"os"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/nexus-rpc/sdk-go/nexusapi"
+)
+
+// Options for creating a Client.
+type Options struct {
+	// Base URL of the service.
+	// Optional. If not provided, this client can only be used to deliver operation completions.
+	ServiceBaseURL string
+	// An HTTP Client to use for making HTTP requests.
+	// Defaults to [http.DefaultClient].
+	HTTPClient *http.Client
+	// Max duration to wait for a single get result request.
+	// Enforced if context deadline for the request is unset or greater than this value.
+	//
+	// Defaults to one minute.
+	GetResultMaxRequestTimeout time.Duration
```

Comment:
Hmmm... I get dropping `Request` but `Max` has value here because this caps the content deadline. The actual request timeout may be lower than this value.

Reply:
How about `GetResultMaxWait`. I mention later about why I think we shouldn't use context deadline as user-provided wait (but we can still kinda bound ourselves by it).